tsl/src/compression/compression.c

```diff
@@ -882,14 +879,14 @@ check_for_limited_size_compressors(PerColumn *pcolumns, int16 natts)
  ** row_compressor **
  ********************/
 void
-row_compressor_init(const CompressionSettings *settings, RowCompressor *row_compressor,
-					Relation uncompressed_table, Relation compressed_table,
-					int16 num_columns_in_compressed_table, bool need_bistate, int insert_options)
+row_compressor_init(RowCompressor *row_compressor, const CompressionSettings *settings,
+					const TupleDesc noncompressed_tupdesc, const TupleDesc compressed_tupdesc)
 {
 	Name count_metadata_name = DatumGetName(
 		DirectFunctionCall1(namein, CStringGetDatum(COMPRESSION_COLUMN_METADATA_COUNT_NAME)));
 	AttrNumber count_metadata_column_num =
-		get_attnum(compressed_table->rd_id, NameStr(*count_metadata_name));
+		get_attnum(settings->fd.compress_relid, NameStr(*count_metadata_name));
+
```

Comment:
I looked over the calls for this one, and they look correct, but since you remove the `compressed_table` from the function parameter, you might want to add an assertion (at least temporarily) before calling `row_compressor_init` making sure that the previous `compressed_table->rd_id` is identical to `settings->fd.compress_relid`. See suggestions.